plugins/com.google.cloud.tools.eclipse.appengine.compat.test/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigratorXsltTest.java

```diff
@@ -84,6 +85,14 @@ public void testWtpMetadataStylesheet_removesGpeRuntimeAndFacets()
     }
   }
 
+  private static Document parseXml(InputStream xml)
+      throws ParserConfigurationException, SAXException, IOException {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
This could shared as a static final field. No need to create a new factory for each request. It's designed to be thread-safe.